### ISSUE TITLE
Injection of parameterized types doesn't work as expected

### ISSUE THREAD

**Issue by MB-LHIND** [avaje/avaje-inject#461]
Hi everyone, I expericene a problem with injecting parameterized types. It seems avaje-inject is not able to find beans, if they have a type parameter even though the bean is present in the bean scope. For example I tried to define a bean of [MongoCollection](https://mongodb.github.io/mongo-java-driver/4.11/apidocs/mongodb-driver-sync/com/mongodb/client/MongoCollection.html) in a `@Factory` annotated configuration class via a `@Bean` annotated factory method:

```java
@Factory
class MyConfig {

    @Bean
    public MongoCollection<MyDocument> buildCollection() {
        // omitted for brevity
    }
}
```

Then I tried to inject the bean into my component test (I tried both using `@InjectTest` as well as programmatically creating the `BeanScope`).

The run fails with: `No bean found for type: ...` even though I can see the bean in the BeanScop, when debugging.

I was able to work around this issue by using the raw type in the definition of the factory method as well as when injecting the bean, but I'd rather not be forced to cast the object everytime:

```java
@Factory
class MyConfig {

    @Bean
    public MongoCollection buildCollection() {
        // omitted for brevity
    }
}
```

Did I miss something? Is this really the intended behavior?

**Comment by SentryMan** [comment]
What's the test look like?

**Comment by SentryMan** [comment]
Ah wait I see the problem, until I fix it you can try the programmatic route like this. 

```java
beanScope.get(new GenericType<MongoCollection<MyDocument>>(){}.type());
```

**Comment by SentryMan** [comment]
fix is in 9.10-RC1
